packages/react-dom/src/client/DOMPropertyOperations.js

```diff
@@ -216,3 +261,11 @@ export function setValueForProperty(
     }
   }
 }
+
+function fireEventProxy(e: Event) {
```

Comment:
Coming back to what I think @sebmarkbage what suggesting — how do you feel about this alternative approach? Am I missing something?

```diff
@ packages/react-dom/src/client/DOMPropertyOperations.js:25 @ import {
  enableCustomElementPropertySupport,
} from 'shared/ReactFeatureFlags';
import {checkAttributeStringCoercion} from 'shared/CheckStringCoercion';
import {getCustomElementEventHandlersFromNode} from './ReactDOMComponentTree';
import {getFiberCurrentPropsFromNode} from './ReactDOMComponentTree';

import type {PropertyInfo} from '../shared/DOMProperty';

@ packages/react-dom/src/client/DOMPropertyOperations.js:169 @ export function setValueForProperty(
    }
    eventName = eventName.slice(2);

    const listenersObjName = eventName + (useCapture ? 'true' : 'false');
    const listeners = getCustomElementEventHandlersFromNode(node);
    const alreadyHadListener = listeners[listenersObjName];
 
    if (typeof value === 'function' || alreadyHadListener) {
      listeners[listenersObjName] = value;
      const proxy = useCapture ? fireEventProxyCapture : fireEventProxy;
      if (value) {
        if (!alreadyHadListener) {
          node.addEventListener(eventName, proxy, useCapture);
        }
      } else {
        node.removeEventListener(eventName, proxy, useCapture);
      }
    const prevProps = getFiberCurrentPropsFromNode(node);
    const prevValue = prevProps != null ? prevProps[name] : null;
    if (typeof prevValue === 'function') {
      node.removeEventListener(eventName, prevValue, useCapture)
    }
    if (typeof value === 'function') {
      node.addEventListener(eventName, value, useCapture);
      return;
    }
  }
@ packages/react-dom/src/client/DOMPropertyOperations.js:258 @ export function setValueForProperty(
  }
}

function fireEventProxy(e: Event) {
  getCustomElementEventHandlersFromNode(this)[e.type + 'false'](e);
}
 
function fireEventProxyCapture(e: Event) {
  getCustomElementEventHandlersFromNode(this)[e.type + 'true'](e);
}
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
modified: packages/react-dom/src/client/ReactDOMHostConfig.js
───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
@ packages/react-dom/src/client/ReactDOMHostConfig.js:445 @ export function commitUpdate(
  newProps: Props,
  internalInstanceHandle: Object,
): void {
  // Apply the diff to the DOM node.
  updateProperties(domElement, updatePayload, type, oldProps, newProps);
  // Update the props handle so that we know which props are the ones with
  // with current event handlers.
  updateFiberProps(domElement, newProps);
  // Apply the diff to the DOM node.
  updateProperties(domElement, updatePayload, type, oldProps, newProps);
}
```

This lets us get rid of the proxies. Note that this shouldn't run at all for memoized handlers because they should get filtered out at `prepareUpdate` level and we shouldn't get into `setsetValueForProperty` for them at all.

Reply:
This is great! I have committed it. I didn't know what getFiberCurrentPropsFromNode does, this is much better!